elfi/classifiers/classifier.py

```diff
@@ -0,0 +1,112 @@
+"""Implementations for ratio estiomation classifiers."""
+
+import abc
+
+import numpy as np
+from sklearn.linear_model import LogisticRegression as LogReg
+from sklearn.preprocessing import StandardScaler
+
+
+class Classifier(abc.ABC):
+    """An abstract base class for a ratio estimation classifier."""
+
+    @abc.abstractmethod
+    def __init__(self):
+        """Initialize a classifier."""
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def fit(self, X, y):
+        """Fit a classifier.
+
+        Parameters
+        ----------
+        X: np.ndarray (n_samples, n_features)
+            Feature vectors of data.
+        y: np.ndarray (n_samples, )
+            Target values, bust be binary.
+
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def predict_log_likelihood_ratio(self, X):
+        """Predict a log-likelihood ratio.
+
+        Parameters
+        ----------
+        X: np.ndarray (n_samples, n_features)
+            Feature vectors of data.
+
+        Returns
+        -------
+        np.ndarray
+
+        """
+        raise NotImplementedError
+
+    def predict_likelihood_ratio(self, X):
+        """Predict a likelihood ratio.
+
+        Parameters
+        ----------
+        X: np.ndarray (n_samples, n_features)
+            Feature vectors of data.
+
+        Returns
+        -------
+        np.ndarray
+
+        """
+        return np.exp(self.predict_log_likelihood_ratio(X))
+
+    @property
+    @abc.abstractmethod
+    def attributes(self):
+        """Return attributes dictionary."""
+        raise NotImplementedError
+
+
+class LogisticRegression(Classifier):
+    """A logistic regression classifier for a ratio estimation."""
+
+    def __init__(self, config=None):
+        """Initialize a logistic regression classifier."""
+        self.config = self._resolve_config(config)
+        self.model = LogReg(**self.config)
+        self.scaler = StandardScaler()
+
+    def fit(self, X, y):
+        """Fit a logistic regression classifier."""
+        Xs = self.scaler.fit_transform(X)
+        self.model.fit(Xs, y)
+
+    def predict_log_likelihood_ratio(self, X):
+        """Predict a log-likelihood ratio."""
+        Xs = self.scaler.transform(X)
+        class_probs = self.model.predict_proba(Xs)[:, 1]
+        return np.log(class_probs / (1 - class_probs))
```

Comment:
add `class_min` to prevent `class_probs=0`

Reply:
If you'd add `class_min`, should you add `class_max` as well? Both based on size of the training set?

Reply:
true we should add `class_max`. it was not needed to make the method work because`class_probs=1`, which would cause a zero division error here, was never observed in our experiments. if i understand correct, `class_probs=0` is observed when the candidate point could not have produced the observed data, whereas `class_probs=1` would be observed when the prior could not have produced the observed data. this makes `class_probs=0` common and `class_probs=1` rare/not expected. however it makes sense to add.

Reply:
Any suggestions to how to adjust both `class_min` and `class_max` based on size of the training set? Further, how to set default values for them?

Reply:
We decided to add `class_min` only at this point.